ts/components/registration/stages/CreateAccount.tsx

```diff
@@ -0,0 +1,157 @@
+    await setSignWithRecoveryPhrase(false);
+    trigger('openInbox');
+  } catch (e) {
+    await resetRegistration();
+    throw e;
+  }
+}
+
+export const CreateAccount = () => {
+  const recoveryPassword = useRecoveryPassword();
+  const displayName = useDisplayName();
+  const displayNameError = useDisplayNameError();
+
+  const dispatch = useDispatch();
+
+  const generateMnemonicAndKeyPair = async () => {
+    if (recoveryPassword === '') {
+      const mnemonic = await generateMnemonic();
+
+      let seedHex = mnDecode(mnemonic);
+      // handle shorter than 32 bytes seeds
+      const privKeyHexLength = 32 * 2;
+      if (seedHex.length !== privKeyHexLength) {
+        seedHex = seedHex.concat('0'.repeat(32));
+        seedHex = seedHex.substring(0, privKeyHexLength);
+      }
+      const seed = fromHex(seedHex);
+      const keyPair = await sessionGenerateKeyPair(seed);
+      const newHexPubKey = StringUtils.decode(keyPair.pubKey, 'hex');
+
+      dispatch(setRecoveryPassword(mnemonic));
+      dispatch(setHexGeneratedPubKey(newHexPubKey)); // our 'frontend' account ID
+    }
+  };
+
+  useMount(() => {
+    void generateMnemonicAndKeyPair();
+  });
+
+  const signUpWithDetails = async () => {
+    if (!(!!displayName && !displayNameError)) {
+      return;
+    }
+
+    try {
+      window.log.debug(
+        `WIP: [onboarding] create account: signUp() is starting display name: ${displayName} recoveryPassword: ${recoveryPassword}`
+      );
+      await signUp({
+        displayName,
+        recoveryPassword,
+      });
+
+      dispatch(setAccountCreationStep(AccountCreation.Done));
+    } catch (e) {
+      window.log.debug(
+        `WIP: [onboarding] create account: creation failed! Error: ${e.message || e}`
+      );
+      dispatch(setAccountCreationStep(AccountCreation.DisplayName));
+      dispatch(setDisplayNameError(e.message || String(e)));
+    }
+  };
+
+  return (
+    <BackButtonWithininContainer
+      margin={'2px 0 0 -36px'}
+      callback={() => {
+        dispatch(setDisplayName(''));
+        dispatch(setRecoveryPassword(''));
+        dispatch(setDisplayNameError(undefined));
+      }}
+    >
+      <Flex
+        container={true}
+        width="100%"
+        flexDirection="column"
+        alignItems="flex-start"
+        margin={'0 0 0 8px'}
+      >
+        <OnboardHeading>{window.i18n('displayNamePick')}</OnboardHeading>
+        <SpacerSM />
+        <OnboardDescription>{window.i18n('displayNameDescription')}</OnboardDescription>
+        <SpacerLG />
+        <SessionInput
+          autoFocus={true}
+          disabledOnBlur={true}
+          type="text"
+          placeholder={window.i18n('enterDisplayName')}
+          value={displayName}
+          onValueChanged={(_name: string) => {
+            const name = sanitizeDisplayNameOrToast(_name, setDisplayNameError, dispatch);
+            dispatch(setDisplayName(name));
+          }}
```

Comment:
usually I use _vars for unused ones
```suggestion
          onValueChanged={(name: string) => {
            const satinizedName = sanitizeDisplayNameOrToast(name, setDisplayNameError, dispatch);
            dispatch(setDisplayName(satinizedName));
          }}
```